cos/static/board_builder.js

```diff
@@ -223,32 +224,35 @@ function update_settlement_color(data,stage,layer) {
 //Illuminates legal settlement locations for placement
 //Legal locations are anywhere without a settlement placed
 function mark_settlement_placement(stage,layer,placed, settlementX, settlementY) {
-    //check for whether there are are remaining settlements
-    var settlements = stage.find('.settlement');
-    var remaining = false;
-    for (i = 0; i < 5; i++){
-        if (settlements[i].x() == settlementX && settlements[i].y() == settlementY){
-            remaining = true;
-            break;
-        }
+    //check if is player's turn
+    if (document.getElementById("is_turn").innerHTML == "true") {
```

Comment:
When the back end returns true for the waitForTurn call, the turn state is updated on the front end by modifying the is_turn element in the game.jinja2 file. That way any UI elements that have to be turned on or off can check if it is the player's turn before initiating any behavior. This was the solution that came to my mind for storing that information on the front end. What would be a better way of going about doing that?

Reply:
I think this could be a dangerous situation as storing a variable that changes often like that could get out of sync with the server. Maybe the UI elements could be set on/off directly in the callback function of the waitForTurn call? I mean, an update UI function can be called from within the callback function, perhaps with the isTurn flag passed if you want to make the function more general.

Reply:
Ok, so then each UI element would have a property attached to it that says whether it's on or off. I could definitely do that. Does that pose the same risks of going out of sync with the server?

Reply:
It shouldn't if the calls from the server are the only things that alter those states. If a call turns a button on, and that button remains active until a call turns it off, it is much more likely to be in sync then switching a global "switch" that could have been turned on or off by a bug in the code or a failure in the response. Also, not that it matters in this project, but it makes things just a bit more secure because it is much more difficult to go through and enable every UI element manually using the browser dev tools than it is to edit the HTML to just turn on that flag. Honestly, both of these situations are unlikely, but they are just things that we have to keep in mind. @davmmeyer, what do you think? Should this move on since we are so far behind?

Reply:
Yes, I think we should move on from this. If it becomes a problem then we can make some type of fix later.